Gems/Terrain/Assets/Shaders/Terrain/TerrainPBR_ForwardPass.azsl

```diff
@@ -107,20 +108,24 @@ ForwardPassOutput TerrainPBR_MainPassPS(VSOutput input)
     surface.position = input.m_worldPosition.xyz;
     surface.vertexNormal = normalize(input.m_normal);
     float viewDistance = length(ViewSrg::m_worldPosition - input.m_worldPosition.xyz);
-    float detailFactor = saturate((viewDistance - TerrainMaterialSrg::m_detailFadeDistance) / max(TerrainMaterialSrg::m_detailFadeLength, EPSILON));
 
     // Surface data to be gathered from clipmaps or materials.
     float3 macroNormal = surface.vertexNormal;
     float3 macroColor = TerrainMaterialSrg::m_baseColor.rgb;
     bool hasDetailSurface = false;
     DetailSurface detailSurface = GetDefaultDetailSurface();
 
+    float detailFactor;
     if (o_useClipmap)
     {
+        // Clipmap has a max render distance. The true fade distance should be the min.
+        float detailFadeDistance = min(TerrainMaterialSrg::m_detailFadeDistance, TerrainSrg::m_clipmapData.m_detailClipmapMaxRenderRadius - TerrainMaterialSrg::m_detailFadeLength);
```

Comment:
This is confusing... The distance that the last detail clipmap renders at should be determined by the detail fade distance, it shouldn't be possible for these to get out of sync.

Reply:
I'll work on an upgrade to expose clipmap settings to TerrainWorldRendererConfig, and sync the 2 values.
This fix will make clipmap visually correct at the fading distance.